BrightID/src/components/NewConnectionsScreens/PreviewConnectionScreen.js

```diff
@@ -27,155 +30,163 @@ import api from '../../Api/BrightId';
  *
  */
 
-type State = {
-  connections: number,
-  groups: number,
-  mutualConnections: number,
-  connectionDate: string,
-  flagged: boolean,
-};
+export const PreviewConnectionScreen = ({ navigation }) => {
+  const dispatch = useDispatch();
 
-export class PreviewConnectionScreen extends React.Component<Props, State> {
-  // eslint-disable-next-line react/state-in-constructor
-  state = {
+  const [userInfo, setUserInfo] = useState({
     connections: 'loading',
     groups: 'loading',
     mutualConnections: 'loading',
     connectionDate: 'loading',
     flagged: false,
-  };
+  });
 
-  componentDidMount() {
-    this.fetchConnectionInfo().done();
-  }
+  const myConnections = useSelector((state) => state.connections.connections);
+  const connectUserData = useSelector(
+    (state) => state.connectUserData,
+    shallowEqual,
+  );
+
+  useFocusEffect(
+    useCallback(() => {
+      if (!connectUserData.photo) {
+        Alert.alert(
+          'Sorry',
+          'There was a problem creating a connection',
+          [
+            {
+              text: 'OK',
+              onPress: reject,
+            },
+          ],
+          { cancelable: true },
+        );
+      }
+      const fetchConnectionInfo = async () => {
+        try {
+          const {
+            createdAt,
+            groups,
+            connections = [],
+            flaggers,
+          } = await api.getUserInfo(connectUserData.id);
+          const mutualConnections = connections.filter(function (el) {
+            return myConnections.some((x) => x.id === el.id);
+          });
+          setUserInfo({
+            connections: connections.length,
+            groups: groups.length,
+            mutualConnections: mutualConnections.length,
+            connectionDate: `Created ${moment(
+              parseInt(createdAt, 10),
+            ).fromNow()}`,
+            flagged: flaggers && Object.keys(flaggers).length > 0,
+          });
+        } catch (err) {
+          if (err instanceof Error && err.message === 'User not found') {
+            setUserInfo({
+              connections: 0,
+              groups: 0,
+              mutualConnections: 0,
+              connectionDate: 'New user',
+              flagged: false,
+            });
+          } else {
+            err instanceof Error ? console.warn(err.message) : console.log(err);
+          }
+        }
+      };
 
-  handleConfirmation = async () => {
-    const { dispatch, navigation } = this.props;
+      fetchConnectionInfo();
+
+      BackHandler.addEventListener('hardwareBackPress', reject);
+      return () => BackHandler.removeEventListener('hardwareBackPress', reject);
+    }, []),
```

Comment:
The useCallback dependencies are missing, should include `connectUserData.id`, `connectUserData.photo`, `myConnections` and also `reject` function. Reject function is tricky - Since it is part of the Component it will change with every render, so making the useCallback useless - It would run on every render call.
You could make reject either a static function defined outside of the component and pass `dispatch` and `navigation` as parameters, or make it a local function inside useCallback. In both approaches `dispatch` and `navigation` need to be added to the callback deps array.

Reply:
I've seen the same warning, however adding this breaks the functionality of this page. Adding the connectUserData variables breaks how we handle what happens when we initially navigate to the page, and we change those variables inside of useCallback. Having zero dependencies inside of useCallback makes the component work as expected. The reason functions like `reject` should exist outside of the component has to do with how functions inside of functional components can hold different values for state or redux, but we're only using dispatch, which will never change.